src/components/IFrame.js

```diff
@@ -42,16 +51,62 @@ export default class IFrame extends React.Component<Props, State> {
     window.addEventListener('message', this.handleMessage);
   }
 
-  componentWillUnmount(){
+  componentWillUnmount() {
     window.removeEventListener('message', this.handleMessage);
   }
 
+  buildSrc = (): string => {
+    const {checkoutId, chargeId, customMetadata} = this.props;
+
+    function encodeURIParams(params) {
+      let encoded = [];
+      let quote = window.encodeURIComponent;
+      for (let key in params) {
+        if (params.hasOwnProperty(key)) {
+          encoded.push(quote(key) + '=' + quote(params[key]));
+        }
+      }
+      return encoded.join('&');
+    }
+
+    let widgetType;
+    let id;
+    if (checkoutId) {
+      id = checkoutId;
+      widgetType = 'checkout';
+    } else if (chargeId) {
+      id = chargeId;
+      widgetType = 'charges';
+    } else {
+      throw new Error('must supply either checkoutId or chargeId prop');
+    }
+
+    const params: SrcParams = {
+      origin: this.hostName,
+      version: VERSION,
+      buttonId: this.uuid,
+    };
+
+    let custom = '';
+    if (customMetadata && typeof customMetadata !== 'string') {
+      console.error('Received customMetadata not of "string" type. Ignoring.');
+    } else if (customMetadata) {
+      custom = customMetadata
+    }
+
+    if (custom){
+      params.custom = custom
+    }
+
+    return `${this.origin}/embed/${widgetType}/${encodeURI(id)}?${encodeURIParams(params)}`;
```

Comment:
nit: i believe that `encodeURI` is more correct as `encodeURIComponent` but I'm not entirely sure. `encodeURI` doesn't encode `?`s or `&`s so a bogus `id` could mess with this URI